syncopy/io/load_ft.py

```diff
@@ -0,0 +1,419 @@
+        keys = select_structures
+    # load all structures found
+    else:
+        keys = struct_keys
+
+    for skey in keys:
+        if skey not in struct_keys:
+            msg = f"Could not find structure `{skey}` in {filename}"
+            SPYWarning(msg)
+            continue
+
+        structure = struct_container[skey]
+        _check_req_fields(req_fields_raw, structure)
+        data = struct_reader(structure)
+        out_dict[skey] = data
+
+    return out_dict
+
+
+def _read_hdf_structure(h5Group,
+                        h5File,
+                        mem_use,
+                        add_fields=None):
+
+    '''
+    Each Matlab structure contained in
+    a hdf5 MAT-File is a h5py Group object.
+
+    Each key of this Group corresponds to
+    a field in the Matlab structure.
+
+    This is the translation from FT to Syncopy:
+
+    FT     Syncopy
+
+    label - channel
+    trial - trial
+    time  - time
+
+    optional:
+    fsample - samplerate
+
+    cfg - X
+
+    '''
+    # for user info
+    struct_name = h5Group.name[1:]
+
+    # this should be fixed upstream such that
+    # the `defaultDimord` is indeed the default :)
+    AData = AnalogData(dimord=AnalogData._defaultDimord)
+
+    # probably better to define an abstract mapping
+    # if we want to support more FT formats in the future
+
+    # these are numpy arrays holding hdf5 object references
+    # e.i. one per trial, channel, time (per trial)
+    trl_refs = h5Group['trial'][:, 0]
+    time_refs = h5Group['time'][:, 0]
+    chan_refs = h5Group['label'][0, :]
+
+    if 'fsample' in h5Group:
+        AData.samplerate = h5Group['fsample'][0, 0]
+    else:
+        AData.samplerate = _infer_fsample(h5File[time_refs[0]])
+
+    # -- retrieve shape information --
+    nTrials = trl_refs.size
+
+    # peek in 1st trial to determine single trial shape
+    # we only support equal trial lengths at this stage
+    nSamples, nChannels = h5File[trl_refs[0]].shape
+    nTotalSamples = nTrials * nSamples
+
+    itemsize = h5File[trl_refs[0]].dtype.itemsize
+    # in Mbyte
+    trl_size = itemsize * nSamples * nChannels / 1e6
+
+    # assumption: single trial fits into RAM
+    if trl_size > 0.4 * mem_use:
+        msg = f"\nSingle trial is at least 40% of the requested chache size of {mem_use}MB\n"
+        msg += f"Still trying to load {trl_size:.1f}MB trials.."
+        SPYWarning(msg)
+
+    # -- IO process --
+
+    # create new hdf5 dataset for our AnalogData
+    # with the default dimord ['time', 'channel']
+    h5FileOut = h5py.File(AData.filename, mode="w")
+    ADset = h5FileOut.create_dataset("data",
+                                     dtype=np.float32,
+                                     shape=[nTotalSamples, nChannels])
+
+    pbar = tqdm(trl_refs, desc=f"{struct_name} - loading {nTrials} trials")
+    SampleCounter = 0   # trial stacking
+
+    # one swipe per trial
+    for tr in pbar:
+        ADset[SampleCounter:nSamples, :] = h5File[tr]
```

Comment:
Is `h5File[tr]` a HDF5 dataset? If so, it could be streamed block by block from disk.

Reply:
It's a dataset containing a single trial, so reading a single trial at once is the strategy we deemed fitting to the rest of the Syncopy architecture?!

Reply:
Got it, my bad - you're right of course!